File: androidApp/src/main/java/dev/emanuelepapa/kmmsurvivalguidesamples/android/coroutines/CoroutinesScreen.kt
package dev.emanuelepapa.kmmsurvivalguidesamples.android.coroutines

import androidx.compose.foundation.layout.Column
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState

@Composable
fun CoroutinesScreen(viewModel: CoroutinesScreenViewModel) {
    val state = viewModel.state.collectAsState().value
    Column {
        Button(onClick = { viewModel.onShowMyNameClicked() }) {
            Text(text = "Show my name!")
        }
        val text = state.formattedName?.let {
            "My name is: $it"
        } ?: "My name is..."
        Text(text = text)
    }
}

File: iosApp/iosApp/coroutines/CoroutinesScreen.swift
//
//  CoroutinesScreen.swift
//  iosApp
//
//  Created by Emanuele on 17/09/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import shared

struct CoroutinesScreen: View {
    
    @ObservedObject var viewModel: CoroutinesScreenViewModel
    
    var body: some View {
        VStack {
            Button {
                viewModel.onShowMyNameClicked()
            } label: {
                Text("Show my name!")
            }
            Text(viewModel.state.formattedName ?? "My name is...")
            Button {
                viewModel.onShowMyNameWithCombineClicked()
            } label: {
                Text("Show my name with async!")
            }
            Text(viewModel.state.formattedNameWithAsync ?? "My name is...")
        }
    }
}
